src/core/forms.py

```diff
@@ -26,9 +25,12 @@ class Meta:
 
 
 class TransactionForm(forms.ModelForm):
+
     def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user')
         super().__init__(*args, **kwargs)
         self.fields['category'].empty_label = "Category not selected"
+        self.fields['category'].queryset = Category.objects.filter(user=self.user)
```

Comment:
Here are the changes from the previous PR, it is not good, so if you start writing new functionality, you need a branch from the actual developer branch (the actual developer branch is dev).